allennlp/common/from_params.py

```diff
@@ -70,6 +71,27 @@ def takes_arg(obj, arg: str) -> bool:
         raise ConfigurationError(f"object {obj} is not callable")
     return arg in signature.parameters
 
+
+def takes_kwargs(obj) -> bool:
+    """
+    Checks whether a provided object takes in any positional arguments.
+    Similar to takes_arg, we do this for both the __init__ function of
+    the classp or a function / method
+    Otherwise, we raise an error
+    """
+    if inspect.isclass(obj):
+        signature = inspect.signature(obj.__init__)
+    elif inspect.ismethod(obj) or inspect.isfunction(obj):
+        signature = inspect.signature(obj)
+    else:
+        raise ConfigurationError(f"object {obj} is not callable")
+    if any([p.kind == inspect._ParameterKind.VAR_KEYWORD  # type: ignore
```

Comment:
pylint has two complaints about this line, and I think it's reasonable to just ignore both of them (with `# pylint: disable=protected-access,simplifiable-if-statement` inside the scope of this function).

Reply:
Changed the inspect to not access the protected variable, and changed the return type to match the simplifiable-if-statement part.